src/System.Diagnostics.TraceSource/src/System/Diagnostics/TraceInternal.cs

```diff
@@ -11,19 +11,44 @@ namespace System.Diagnostics
 {
     internal static class TraceInternal
     {
+        static TraceInternal()
+        {
+            // This is where we override default DebugProvider because we know
+            // for sure that we have some Listeners to write to.
```

Comment:
This comment does not seem to hold anymore. I think the `Debug.SetProvider(new TraceProvider())` line can be where it used to be before.

Reply:
The reason I need this in static constructor is because even when someone calls Trace.IndentLevel (getter/setters) I want this statement to get called (not only when they call Listeners).

I'll update/remove the comment

Reply:
(Actually, it should be first in `Listeners` property - so that we start receiving the events before hooking up TraceListenerCollection.)